src/App.tsx

```diff
@@ -1,26 +1,49 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.css';
 import { getNumbers } from './utils';
+import { Pagination } from './components/Pagination';
 
 // eslint-disable-next-line @typescript-eslint/no-unused-vars
-const items = getNumbers(1, 42)
+const items: string[] = getNumbers(1, 42)
   .map(n => `Item ${n}`);
 
 export const App: React.FC = () => {
+  const [currentPage, setCurrentPage] = useState(1);
+  const [itemsPerPage, setItemsPerPage] = useState(5);
+
+  const totalPages = items.length;
+  const firstVisibleItem = (currentPage * itemsPerPage) - itemsPerPage;
+  const lastVisibleItem = (firstVisibleItem + itemsPerPage - 1) <= totalPages
+    ? firstVisibleItem + itemsPerPage
+    : totalPages;
+
+  const handleChangePage = (newPage: number) => {
+    if (newPage !== currentPage) {
+      setCurrentPage(newPage);
+    }
+  };
+
+  const visibleArray = items.slice(firstVisibleItem, lastVisibleItem);
+
   return (
     <div className="container">
       <h1>Items with Pagination</h1>
 
       <p className="lead" data-cy="info">
-        Page 1 (items 1 - 5 of 42)
+        {`Page ${currentPage} (items ${firstVisibleItem + 1} - ${lastVisibleItem} of ${totalPages})`}
       </p>
 
       <div className="form-group row">
         <div className="col-3 col-sm-2 col-xl-1">
           <select
             data-cy="perPageSelector"
+            value={itemsPerPage}
             id="perPageSelector"
             className="form-control"
+            onChange={(event) => {
+              setItemsPerPage(+(event.target.value));
+              setCurrentPage(1);
+            }}
```

Comment:
Move to separate function  (transfer the function to a variable, and here only write the name of the function)